### PR TITLE
Fix: make getScope acquire innermost scope (fixes #3700)

### LINKED ISSUE #3700: getScope should retrieve module scope instead of global scope for ES6 modules
escope separates `global` scope from `module` scope for ES6 modules by [nesting a `module` scope within the `global` scope](https://github.com/estools/escope/blob/45cf8512857ac5663866d2a08d1d8fb28f78366f/src/referencer.js#L496-l498). `getScope` currently [`acquire`s](https://github.com/eslint/eslint/blob/fe3a047b6b29e97597c8e8f9882f168c17e15a08/lib/eslint.js#L967) [the outermost scope](https://github.com/estools/escope/blob/6a45f68c9aa3937dcb37fe177932683b84892b9a/src/scope-manager.js#L95-L102) from escope. This leads to the `global` scope being returned when `getScope` is called in context of a top level statement instead of the `module` scope.

### PR DESCRIPTION
**Pull request by voideanvalue**
I'm not sure if this is the best way to do it, but it fixes #3700 and doesn't break anything else. I've also added a unit test that fails without this change and passes with.

### CODE REVIEW

**Comment by IanVS** [comment]
Thanks for the PR, but it's best to wait until your issue is marked as "accepted" before submitting.

Also, we have some unique rules for how commits must be named.  Please see http://eslint.org/docs/developer-guide/contributing/pull-requests#step-2-make-your-changes for information on how to make the commit-check test pass.

**Comment by voideanvalue** [comment]
@IanVS: Thanks for the feedback! Part of the reason I submitted the PR immediately is because the unit test highlights the issue, but I guess I could've also just linked to my fork.

I'll tag with fix and leave this here for now. I don't think it's a breaking change since it doesn't break any of the existing plugins in the repository, but if anyone disagrees with that... let me know!

**Comment by IanVS** [comment]
No problem.  The reason we ask contributors to wait until accepted is it gives the team a chance to review the issue and discuss/guide possible solutions before work is spent on a fix.

File: tests/lib/eslint.js [hunk 1]
```diff
@@ -564,6 +564,18 @@ describe("eslint", function() {
 
             eslint.verify("try {} catch (err) {}", config, filename, true);
         });
+
+        it("should retrieve module scope correctly from an ES6 module", function() {
```
Comment (nzakas) [inline tests/lib/eslint.js]: Can you add a test when using globalReturn as well?

Comment (voideanvalue) [inline tests/lib/eslint.js]: Sure, done!

**Comment by ilyavolodin** [comment]
LGTM

**Pull request merged**